### PR TITLE
Delete some broken and unused SparsePoly code

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by thofma** [comment]
Has @fieker any comment?

File: src/generic/SparsePoly.jl [hunk 1]
```diff
@@ -610,154 +329,6 @@ end
...
-            reuse = xn
-         end
-         if v.j < gnext - 1 && (largest[v.i] & mask) <  v.j + 1
-            if reuse != 0
-               I[reuse] = heap_t(v.i, v.j + 1, 0)
-               heapinsert!(H, I, reuse, f.exps[v.i] + ge[v.j + 1]) # either chain or insert v into heap
-               reuse = 0
-            else
-               push!(I, heap_t(v.i, v.j + 1, 0))
-               Collections.heappush!(H, heap_sr(f.exps[v.i] + ge[v.j + 1], length(I)))
```
Comment (fingolfin) [inline src/generic/SparsePoly.jl:729]: To be explicit, this code is broken because there is no `Collections.heappush!`

**Comment by fingolfin** [comment]
Before merging this, we should make sure the `evaluate` and `gcd` code for MPolys which is using SparsePoly still works -- right now I am not even sure *those* are tested... If not, we should add tests. I'll look into this, until then, this should not be merged.

**Comment by fingolfin** [comment]
OK I verified that the `evaluate` and `gcd` methods using `SparsePoly` both are covered by our test suite and do *not* make use of the code being deleted here.

**Pull request merged**